src/System.Drawing.Common/src/System/Drawing/Icon.Windows.cs

```diff
@@ -374,7 +374,46 @@ private void DrawIcon(IntPtr dc, Rectangle imageRect, Rectangle targetRect, bool
             }
             finally
             {
-                SafeNativeMethods.RestoreClipRgn(dc, hSaveRgn);
+                RestoreClipRgn(dc, hSaveRgn);
+            }
+        }
+
+        private static IntPtr SaveClipRgn(IntPtr hDC)
+        {
+            IntPtr hTempRgn = Interop.Gdi32.CreateRectRgn(0, 0, 0, 0);
+            IntPtr hSaveRgn = IntPtr.Zero;
+            try
```

Comment:
The `try..finally` here isn't useful. There would have to be some catastrophic failure (no memory maybe) and at that point a dangled GDI object isn't a terrible thing, if we could manage to delete it anyway.